crates/ruff_python_parser/src/parser/mod.rs

```diff
@@ -533,54 +528,95 @@ impl<'src> Parser<'src> {
             .recovery_context
             .union(RecoveryContext::from_kind(recovery_context_kind));
 
+        let mut first_element = true;
         let mut trailing_comma_range: Option<TextRange> = None;
 
         loop {
             progress.assert_progressing(self);
 
-            // The end of file marker ends all lists.
-            if self.at(TokenKind::EndOfFile) {
-                break;
-            }
-
             if recovery_context_kind.is_list_element(self) {
                 parse_element(self);
 
+                // Only unset this when we've completely parsed a single element.
+                if first_element {
+                    first_element = false;
+                }
+
```

Comment:
Nit: We can always unset it. Checking that it's true isn't realy necessary. All that matters here is that it's always `false` after. 

```suggestion
                
                first_element = false;

```

what I would find more interesting in the comment is the reason why we should only unset it when we've completely parsed the first element.